langchain-core/src/utils/stream.ts

```diff
@@ -98,3 +98,63 @@ export class IterableReadableStream<T>
     });
   }
 }
+
+export function atee<T>(
+  iter: AsyncGenerator<T>,
+  length = 2
+): AsyncGenerator<T>[] {
+  const buffers = Array.from(
+    { length },
+    () => [] as Array<IteratorResult<T> | IteratorReturnResult<T>>
+  );
+  return buffers.map(async function* makeIter(buffer) {
+    while (true) {
+      if (buffer.length === 0) {
+        const result = await iter.next();
+        for (const buffer of buffers) {
+          buffer.push(result);
+        }
+      } else if (buffer[0].done) {
+        return;
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        yield buffer.shift()!.value;
+      }
+    }
+  });
+}
+
+export function concat<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends Array<any> | string | number | Record<string, any> | any
+>(first: T, second: T): T {
+  if (Array.isArray(first) && Array.isArray(second)) {
+    return first.concat(second) as T;
+  } else if (typeof first === "string" && typeof second === "string") {
+    return (first + second) as T;
+  } else if (typeof first === "number" && typeof second === "number") {
+    return (first + second) as T;
```

Comment:
This seems like an odd thing to do

Reply:
This is what we do in py

Reply:
Streaming number would be a bit weird in general I guess